src/zope/interface/_zope_interface_coptimizations.c

```diff
@@ -400,7 +400,7 @@ static PyMemberDef SB_members[] = {
     { "__iro__", T_OBJECT_EX, offsetof(SB, __iro__), 0, "" },
     { "__sro__", T_OBJECT_EX, offsetof(SB, __sro__), 0, "" },
 #if USE_EXPLICIT_WEAKREFLIST
-    { "__weaklistoffset__", T_OBJECT_EX, offsetof(SB, weakreflist), 0, "" },
+    { "__weaklistoffset__", T_PYSSIZET, offsetof(SB, weakreflist), READONLY, "" },
```

Comment:
Seems good, following the description for the `__vectorcalloffset__` member [here](https://docs.python.org/3/c-api/structures.html#c.PyMemberDef).